streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -570,13 +570,14 @@ private void captureOutputRecords() {
 
             // Forward back into the topology if the produced record is to an internal or a source topic ...
             final String outputTopicName = record.topic();
-            final long timestamp = Objects.requireNonNull(record.timestamp());
             if (internalTopics.contains(outputTopicName)
                 || processorTopology.sourceTopics().contains(outputTopicName)) {
 
                 final TopicPartition topicPartition = getTopicPartition(record.topic());
+                final long timestamp = Objects.requireNonNull(record.timestamp());
                 enqueueTaskRecord(record.topic(), topicPartition, timestamp, record.key(), record.value(), record.headers());
             } else if (globalPartitionsByTopic.containsKey(outputTopicName)) {
+                final long timestamp = Objects.requireNonNull(record.timestamp());
```

Comment:
`ProducerRecord` is actually guarantee to have a non-null timestamp -- not sure if we need this check -- actually similar for other fields like topic or header